config/v1/types_authentication.go

```diff
@@ -50,6 +50,10 @@ type AuthenticationSpec struct {
 	// The namespace for these secrets is openshift-config.
 	// +optional
 	WebhookTokenAuthenticators []WebhookTokenAuthenticator `json:"webhookTokenAuthenticators,omitempty"`
+
+	// serviceAccountIssuer is the identifier of the bound service account token issuer.
+	// +optional
+	ServiceAccountIssuer string `json:"serviceAccountIssuer"`
```

Comment:
is there any special format? Is it a URL ?

Reply:
Is there a default value?

Reply:
maybe `serviceAccountIssuer configures the issuer for bound service account tokens` would make more sense as an explanation for a common user?

Reply:
> It's just a string. It can be configured with a URL to source the public key to verify issued tokens (as is the case for the EKS webhook), but it can also just be a textual value used by the verifier in kube-apiserver to identify that a token was issued by it.

Provide the recommended options in the godoc. 



> As to the question of a default value, the operator will set `auth.openshift.io` in [defaultconfig.yaml](https://github.com/openshift/cluster-kube-apiserver-operator/pull/718/commits/86d73cef4483a553ea6d43c3b5c677d0e70f076d#diff-b09019334d31bbad7ae1cacca007940dR110). Does that need to be documented here given that the API isn't the one defaulting it?

Yes, the behavior of the system and the repercussions of choices needs to be described here.

Reply:
Do we know of existing use-cases that require a non-url string?  If not, let's start by having it be a URL only, but specify in the doc that consumers of this API *must* tolerate any string.

Reply:
afaict providing a issuer URL in a token is an indication that public keys can be retrieved from said URL. iirc you made explicit mention of not wanting to provide a valid URL. Is providing an invalid URL preferable to providing a non-URL?

Reply:
Is every UTF-8 string acceptable?

Reply:
Do we handle empty string correctly? Should we set omitempty?

Reply:
As to empty string, I'll have to get back to you. I don't have a test server handy after the one I was using died.

Reply:
> Should I be duplicating validation that the apiserver will perform when it sees the arg? I'm assuming that would involve a webhook given that this is a CRD and afaik that kind of logic can't be expressed as a regular expression.

Add a OpenAPI pattern here, i.e. a regex. I would also set the minimum length to 1. No need to allow such a corner case of an empty string.

Reply:
re: empty string, my intent is to use the default value if an empty string is set here. Does that imply setting omitempty or not?

Reply:
something like `[^:]*|someURLRegEx`

Reply:
Wait, why don't I want to allow an empty string given that will be the default?

Reply:
We could. But then we have two ways to express that: empty and undefined. Kind of ugly.

Reply:
A 'url regex' is a non-trivial thing (as per https://mathiasbynens.be/demo/url-regex). I get that it's desirable to validate up-front, but having to validate complex data declaratively like this is error prone to say the least. How about doing a url check in code when the field is observed and reporting an error in the operator logs or an event?

Reply:
am fine with later validation and reporting via e.g. a condition